File: src/main/scala/io/koju/autopos/transaction/api/ReceiptApi.scala
package io.koju.autopos.transaction.api

import io.koju.autopos.kernel.api.BaseApi
import io.koju.autopos.transaction.domain.Receipt
import io.koju.autopos.transaction.repo.ReceiptRepo
import org.springframework.web.bind.annotation.{RequestMapping, RestController}

@RestController
@RequestMapping(Array("/api/receipts"))
class ReceiptApi(private val receiptRepo: ReceiptRepo)
  extends BaseApi[Receipt](receiptRepo, "receipt", "/api/receipts")

File: src/main/kotlin/com/kapilkoju/autopos/transaction/api/ReceiptApi.kt
package com.kapilkoju.autopos.transaction.api

import com.kapilkoju.autopos.kernel.api.BaseApi
import com.kapilkoju.autopos.transaction.domain.Receipt
import com.kapilkoju.autopos.transaction.service.ReceiptRepo
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(ReceiptApi.baseUrl)
class ReceiptApi(private val receiptRepo: ReceiptRepo)
    : BaseApi<Receipt>(receiptRepo, "receipt", baseUrl) {

    companion object {
        const val baseUrl = "/api/receipts"
    }
}
